src/tools/dataset.cpp

```diff
@@ -1197,38 +1202,31 @@ void ustore_docs_import(ustore_docs_import_t* c_ptr) {
     if (!c.arena)
         c.arena = arena_t(c.db).member_ptr();
 
+    auto arena = linked_memory(c.arena, c.options, c.error);
+    return_if_error_m(c.error);
+
+    auto handle_exception = [&](char const* ex) {
+        auto ptr = arena.alloc<char>(strlen(ex), c.error);
+        std::memcpy(ptr.data(), ex, strlen(ex) + 1);
```

Comment:
What if we run out of memory? This SEGFAULTs

Reply:
Fixed.